luatest/server_v2.lua

```diff
@@ -0,0 +1,503 @@
+--
+-- @return table
+function Server:build_env()
+    local res = getmetatable(getmetatable(self)).build_env(self)
+    if self.box_cfg ~= nil then
+        res.TARANTOOL_BOX_CFG = json.encode(self.box_cfg)
+    end
+    return res
+end
+
+--- Build a listen URI based on the given server alias.
+-- For now, only UNIX sockets are supported.
+--
+-- @string alias Server alias.
+-- @return string
+function Server.build_listen_uri(server_alias)
+    return fio.pathjoin(Server.vardir, server_alias .. '.sock')
+end
+
+--- Start a server.
+-- Waits until the server is ready, unlike the 1st version of the server class.
+--
+-- @tab[opt] opts
+-- @bool[opt] opts.wait_until_ready Wait until the server is ready.
+--   Defaults to `true`.
+function Server:start(opts)
+    checks('table', {wait_until_ready = '?boolean'})
+    getmetatable(getmetatable(self)).start(self)
+    local wait_until_ready = true
+    if opts ~= nil and opts.wait_until_ready ~= nil then
+        wait_until_ready = opts.wait_until_ready
+    end
+    if wait_until_ready then
+        self:wait_until_ready()
+    end
+end
+
+-- TODO: Add Server:restart() function.
+
+--- Stop the server.
+-- Waits until the server process is killed, unlike the 1st version of the
+-- server class.
+function Server:stop()
+    if self.process then
+        local pid = self.process.pid
+        getmetatable(getmetatable(self)).stop(self)
+        local deadline = clock.time() + WAIT_TIMEOUT
+        while true do
+            if ffi.C.kill(pid, 0) ~= 0 then
+                break
+            end
+            if clock.time() > deadline then
+                error(('Stopping of server (alias: %s, workdir: %s, pid: %d) timed out')
+                    :format(self.alias, fio.basename(self.workdir), pid))
+            end
+            fiber.sleep(WAIT_DELAY)
+        end
+    end
+end
+
+--- Clean the server's working directory.
+-- Should be invoked only for the stopped server.
+function Server:clean()
+    fio.rmtree(self.workdir)
+    self.instance_id = nil
+    self.instance_uuid = nil
+end
+
+--- Stop the server and clean its working directory.
+function Server:drop()
+    self:stop()
+    self:clean()
+end
+
+-- Wait until the given condition is `true` (anything except `false` and `nil`).
+-- Throw an error when timeout exceeds.
+local function wait_for_condition(cond_name, server, func, ...)
+    local pid = server.process.pid
+    local deadline = clock.time() + WAIT_TIMEOUT
+    while true do
+        if func(...) then
+            return
+        end
+        if clock.time() > deadline then
+            error(('Waiting for "%s" condition on server (alias: %s, workdir: %s, pid: %d) timed out')
+                :format(cond_name, server.alias, fio.basename(server.workdir), pid))
+        end
+        fiber.sleep(WAIT_DELAY)
+    end
+end
+
+--- Wait until the server is ready after the start.
+-- It is considered that the server is operable when its `_G.ready` variable
+-- becomes `true`.
```

Comment:
brings meaningful words together on one line

```suggestion
-- A server is considered operable when its
-- `_G.ready` variable becomes `true`.
```